### PR TITLE
Fixes 

### PR DESCRIPTION
**Pull request by cuanl-cd**
This branch includes various small fixes for the pre-requisite deployment.

### CODE REVIEW

**Comment by richeney** [comment]
This looks like it is merging two repos together.

**Comment by richeney** [comment]
If I am renaming @StuartCD's repo content to terraform-morpheus-bootstrap then it looks like that should be the 1.0.0 area. This will be run by either a PS consultant or by the customer. Therefore we don't want the 2.0.0 content in a public repo.

**Comment by cuanl-cd** [comment]
Hi @richeney I have updated the code to remove the various layers and only include the "first layer" terraform module. I have also made some amendments to the variable naming.

File: main.tf [hunk 1]
```diff
@@ -103,25 +116,36 @@ resource "azurerm_role_assignment" "key_vault_rbac_current_user" {
...
+  value        = var.cd_github_pat_value
   key_vault_id = azurerm_key_vault.KeyVault.id
+
+  depends_on = [azurerm_role_assignment.key_vault_rbac_current_user]
 }
 
 //Create the provider file for the next module
 
 resource "local_file" "terraform_providers" {
   filename = "${path.root}/../2.0.0.CDMonitoring_Prerequisite_Deployment/providers.tf"
```
Comment (richeney) [inline main.tf:144]: @cuanl-cd perhaps a templates subdir and add these two files - providers.tf and terraform.tfvars - into there?

File: terraform.tfvars.example [hunk 2]
```diff
@@ -15,5 +18,7 @@ org_tag         = "Cloud Direct"
 
 // Variables for providers.tf template creation
 
-devops_org_service_url = ""
-
+devops_org_service_url = "https://dev.azure.com/clouddirect"
+cust_subscription_name = ""
+cust_tennant_id        = ""
```
Comment (richeney) [inline terraform.tfvars.example:23]: @cuanl-cd change all instances of tenannt to tenant?

File: variables.tf [hunk 3]
```diff
@@ -1,6 +1,18 @@
...
-variable "subscription_id" {
+variable "cust_subscription_id" {
+  description = "Specify the Azure Subscription to deploy into."
+}
+
+variable "cust_subscription_name" {
+  description = "Specify the Azure Subscription to deploy into."
+}
+
+variable "cust_tennant_name" {
```
Comment (richeney) [inline variables.tf:11]: tennant

**Review by richeney — CHANGES_REQUESTED** [review]
@cuanl-cd I added a few comments inline

File: main.tf (hunk 1 above)
Comment (cuanl-cd) [reply main.tf:144]: have added templates directory for template files

File: variables.tf (hunk 3 above)
Comment (cuanl-cd) [reply variables.tf]: corrected typos

File: main.tf (hunk 1 above)
Comment (richeney) [reply main.tf:144]: Does line 144 need to specify it? E.g.

```terraform
filename = "${path.root}/templates/providers.tf"
```

File: main.tf [hunk 4]
```diff
@@ -103,25 +116,36 @@ resource "azurerm_role_assignment" "key_vault_rbac_current_user" {
...
   content  = local.terraform_providers
-}
\ No newline at end of file
+}
+
+resource "local_file" "terraform_tfvars" {
+  filename = "${path.root}/../2.0.0.CDMonitoring_Prerequisite_Deployment/terraform.tfvars"
```
Comment (richeney) [inline main.tf:149]: And same again for this?

File: variables.tf [hunk 5]
```diff
@@ -43,3 +63,7 @@ variable "description_tag" {
 variable "org_tag" {
   description = "Specify the Org Tag"
 }
+
+variable "cd_github_org_name" {
+  description = "Specify the Github Orginization"
```
Comment (richeney) [inline variables.tf:68]: typo

**Review by richeney — CHANGES_REQUESTED** [review]
Almost there...

File: main.tf (hunk 4 above)
Comment (cuanl-cd) [reply main.tf:149]: It wont need to reference the templates folder as the location is provided by content which in turn is sourced from the terraform_providers and terraform_tfvars locals respectively

**Review by richeney — APPROVED** [review]

**Comment by richeney** [comment]
Will merge in the PR and then we can look at the GitHub provider resources - e.g. data for the repo, which should be copied from a template repo. We can then have the fine grained PAT created for that repo, and push up the required files once the GitHub Actions have been configured for the workload identity.

**Pull request merged**